engine/execution/ingestion/stop_control_test.go

```diff
@@ -0,0 +1,73 @@
+package ingestion
+
+import (
+	"context"
+	"testing"
+
+	"github.com/onflow/flow-go/engine/execution/state/mock"
+	"github.com/rs/zerolog"
+	mock2 "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// If stopping mechanism has caused any changes to execution flow (skipping execution of blocks)
+// we disallow setting new values
+func TestCannotSetNewValuesAfterStoppingCommenced(t *testing.T) {
+
+	t.Run("when processing block at stop height", func(t *testing.T) {
+		sah := NewStopControl(zerolog.Nop(), false)
```

Comment:
useful to also have log in tests in case we need to debug
```suggestion
		sah := NewStopControl(unittest.Logger(), false)
```

Reply:
👍